react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+
+class Images extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = { count: 1 };
+    this.count = this.count.bind(this);
+    this.advance = this.advance.bind(this);
+    this.backwards = this.backwards.bind(this);
+    this.iconClick = this.iconClick.bind(this);
+
+  }
+
+  count() {
+    let increase = this.state.count;
+    if (increase < 5) {
+      increase++;
+      this.setState({ count: increase });
+    } else {
+      increase = 1;
+      this.setState({ count: increase });
+    }
+  }
+
+  iconClick(event) {
+
+    const pictureCount = event.target.getAttribute('iconid');
+    this.setState({ count: Number(pictureCount) });
+  }
+
+  advance() {
+
+    let increase = this.state.count;
+    if (increase < 5) {
+      increase = increase + 1;
+      this.setState({ count: increase });
+    } else {
+      increase = 1;
+      this.setState({ count: increase });
+    }
+  }
+
+  backwards() {
+
+    let backcount = this.state.count;
+
+    if (backcount === 1) {
+      backcount = 5;
```

Comment:
also here. we want to avoid having to change our code if our input (props) changes.